src/runtime_src/core/pcie/windows/shim.cpp

```diff
@@ -553,18 +553,22 @@ struct shim
     DWORD error = 0;
     DWORD bytesWritten;
     size_t off = 0;
-    size_t len = 0;
+    size_t ksize = 0;
+    PXOCL_READ_AXLF_ARGS axlf_obj = nullptr;
 
     auto top = reinterpret_cast<const axlf*>(ImageBuffer);
 
-    XOCL_READ_AXLF_ARGS axlf_obj = {0};
-
     auto kernels = xrt_core::xclbin::get_kernels(top);
     /* Calculate size of kernels */
     for (auto& kernel : kernels) {
-        axlf_obj.ksize += sizeof(kernel_info) + sizeof(argument_info) * kernel.args.size();
+        ksize += sizeof(kernel_info) + sizeof(argument_info) * kernel.args.size();
```

Comment:
should we do -1 for the sake of completion?

Reply:
Yes.. fixed